Reloaded.Imgui.Hook.OpenGL/ImguiHookOpenGL.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using DearImguiSharp;
+using Reloaded.Hooks.Definitions;
+using Reloaded.Imgui.Hook.Implementations;
+using Reloaded.Imgui.Hook.OpenGL;
+using static Reloaded.Imgui.Hook.Misc.Native;
+using Debug = Reloaded.Imgui.Hook.Misc.Debug;
+
+namespace Reloaded.Imgui.Hook.Implementations
+{
+    public unsafe class ImguiHookOpenGL : IImguiHook
+    {
+        public static ImguiHookOpenGL Instance { get; private set; }
+
+        private IHook<OpenGLHook.WglSwapBuffers> _wglSwapBuffersHook;
+        private IHook<OpenGLHook.SwapBuffers> _swapBuffersHook;
+        private bool _initialized = false;
+        private bool _wglUsed = false;
+        private bool _gdiUsed = false;
+        private IntPtr _windowHandle;
+        private IntPtr _deviceContext;
+
+        /*
+         * In some cases (E.g. with Viewports enabled), Dear ImGui might call
+         * OpenGL functions from within its internal logic.
+         *
+         * We put a lock on the current thread in order to prevent stack overflow.
+         */
+        private bool _wglSwapBuffersRecursionLock = false;
+        private bool _swapBuffersRecursionLock = false;
+        private bool _isRendering = false;
+
+        public ImguiHookOpenGL() { }
+
+        public void Initialize()
+        {
+            Instance = this;
+
+            ImGui.GetIO().SetPlatformImeDataFn = (viewport, data) =>
```

Comment:
Leftover dummy here

Reply:
Oops! Sorry about that! Working on it now

Reply:
I think this might be needed due to https://github.com/ocornut/imgui/issues/5535. Would it be better to remove this code or add comments explaining why it's here?

Reply:
THERE'S A THIRD ONE?! This is also me realizing you created this _and_ Reloaded. That's awesome! I want to say I think you've done something really great with this framework. It's so much better than what came before, like Easyhook.

Also, I've just pushed a commit that adds a comment for this.